### PR TITLE
Scala.js support via crossbuild

### LINKED ISSUE #3: Scala.js build
Is it something you consider?  I can probably work on a PR due to our need.

### PR DESCRIPTION
**Pull request by benhutchison**
Fixes #3. Using Full crossbuild as described [here](https://www.scala-js.org/doc/project/cross-build.html), so library sources go in `shared/`, and JVM specific sources (eg test, bench) go under `jvm`. Specs and benchmark only runs in jvm project.

Verification progress:
- [x] Build compiles and loads
- [x] `effJVM/test` is green
- [x] `effJS/publishLocal` runs and produces `eff-cats_sjs0.6_2.11 to /Users/ben/.ivy2/local/org.atnos/eff-cats_sjs0.6_2.11/1.4-20160430150555-d069eac/jars/eff-cats_sjs0.6_2.11.jar`
- [x] `effJVM/publishLocal` runs and produces `published eff-cats_2.11 to /Users/ben/.ivy2/local/org.atnos/eff-cats_2.11/1.4-20160430150555-d069eac/jars/eff-cats_2.11.jar`
- [x] Manual test built artifact in JVM Eff example
- [x] Manual test built artifact in JS Eff example
- [ ] Check if bench commands work in `effJVM`

### CODE REVIEW

**Comment by etorreborre** [comment]
Wow, that's amazing! This + the Scala 2712 plugin will be really good.

File: project/depend.scala [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
 
 object depend {
 
   val catsVersion     = "0.4.0"
   val specs2Version   = "3.6.5"
 
   val cats = Seq(
-      "org.typelevel" %% "cats"
+      "org.typelevel" %%%! "cats"
```
Comment (benhutchison) [inline project/depend.scala]: plain `%%%` doesnt work, its not well documented but some context at https://github.com/vmunier/play-with-scalajs-example/issues/20

**Comment by benhutchison** [comment]
Generated artefact tested ok on both platforms, in this testbed project:
https://github.com/benhutchison/eff-play-scalajs/blob/master/build.sbt#L15

ie `epJVM/run` && `epJS/run` both successful and same output

IMO this PR ready to be considered for merge

**Comment by benhutchison** [comment]
I cant seem to run `bench:compile` even on master branch, guessing it works over in `eff-scalaz`?:
https://gist.github.com/benhutchison/0b3b25ae6962611f087c99d1649d7b61

**Comment by etorreborre** [comment]
@benhutchison I haven't forgotten this PR. I'll try to rebase it on master and merge it before the end of the week.

**Comment by benhutchison** [comment]
Great! but BTW  no hurry from my POV, im running off local scala.js builds
just fine.

**Comment by ngbinh** [comment]
I would love to see an official release with scala.js support. Local build is no fun! :)

**Comment by etorreborre** [comment]
@benhutchison @ngbinh I have now merged everything on master and I published some artefacts for version `1.5.1-20160508090222-541f0a6`. Please tell me if that works for you and thanks again for the PR.

**Comment by ngbinh** [comment]
Thanks! Works like charm!

**Pull request merged**